lib/core/test/unit/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobinSpec.hs

```diff
@@ -1610,29 +1741,57 @@ isValidMakeChangeData p = (&&)
     (totalOutputValue `leq` totalInputValue)
     (totalOutputCoinValue > Coin 0)
   where
-    totalInputValue = TokenBundle.add
-        (F.fold $ inputBundles p)
-        (maybe TokenBundle.empty TokenBundle.fromCoin (view #extraCoinSource p))
-    totalOutputValue = F.fold $ outputBundles p
+    totalInputValue =
+        F.fold (inputBundles p)
+            <> (F.foldMap TokenBundle.fromCoin (view #extraCoinSource p))
+            <> TokenBundle.fromTokenMap (view #assetsToMint p)
+    totalOutputValue =
+        F.fold (outputBundles p)
+            <> TokenBundle.fromTokenMap (view #assetsToBurn p)
     totalOutputCoinValue = TokenBundle.getCoin totalOutputValue
 
 genMakeChangeData :: Gen MakeChangeData
 genMakeChangeData = flip suchThat isValidMakeChangeData $ do
     outputBundleCount <- choose (0, 15)
     let inputBundleCount = outputBundleCount * 4
+
+    inputBundles <- genTokenBundles inputBundleCount
+    outputBundles <- genTokenBundles outputBundleCount
+
+    -- Tokens in the input but not the output can be burned
+    assetsToBurn <- TokenMap.fromFlatList <$>
```

Comment:
As discussed with @jonathanknowles , I do not think these generators sufficiently cover the potential "universe" of MakeChangeData. They unnecessarily make the set of user-specified assets and the set of assets to burn disjoint.

I made a hacky PR to see how the tests would behave if I altered the generator: https://github.com/input-output-hk/cardano-wallet/pull/2740

Reply:
Hi @sevanspowell 

I've added some extra coverage checks in 96b6a756f6217e5801ff3c647d8f36d1c30dd222.

It's not as bad as we expected: we do get some overlap between the sets of user specified assets and the sets of assets to burn.

Though we were missing coverage for the case where there's an overlap between the sets of assets to mint and the assets to burn. So I've tweaked the generator a bit to allow for this case in 378d35f5b37097e8814ced6801e5988213654a96.